crates/sui-framework/packages/sui-framework/tests/random_tests.move

```diff
@@ -103,6 +106,9 @@ module sui::random_tests {
             scenario.ctx(),
         );
 
+        // Deterministically set tx context so the RNG seeded by the ctx below is deterministic.
```

Comment:
Here, we're kind of hoping that someone doesn't create an object using this ctx in later code, right? Because if they did, we could be in that bad situation where two tests run in parallel and write different contents for the same object.

FWIW, I preferred the explicit use of the ctx you get from `new_from_hint` instead of overwriting the scenario's context for that reason.